packages/client/src/Components/Mapping/helpers/routeLinesFor.js

```diff
@@ -0,0 +1,302 @@
+  /* boolean */ lightweight, /* grid */ grid, /* string */ color, /* int */ planningFor, /* boolean */ highlight, /* boolean */ solid) {
+  // note - we will actually start with a layer group, in case we're showing
+  // a bold line and a feint line
+  const res = L.layerGroup()
+
+  let weight
+  if (highlight) {
+    // just make it a bold track
+    weight = 4
+  } else {
+    weight = lightweight && !planningFor ? 1 : 2
+  }
+  const dashArray = solid ? [] : lightweight ? [1, 7] : [4, 8]
+  const boldLine = L.polyline([], {
+    dashArray: dashArray,
+    weight: weight,
+    color: color
+  })
+  const feintLine = L.polyline([], {
+    dashArray: dashArray,
+    weight: 2,
+    color: color
+  })
+
+  if (turns) {
+    const boldLatLongs = [start]
+    const feintLatLongs = []
+    let lastPos = null
+    turns.forEach(turn => {
+      const list = planningFor && planningFor >= turn.turn ? boldLatLongs : feintLatLongs
+      // loop through the routes
+      if (turn.route) {
+        // loop through the steps in this route
+        turn.route.forEach(step => {
+          const ptHex = grid.hexNamed(step)
+          if (ptHex) {
+            const location = ptHex.centrePos
+            // is this the first feint line?
+            if (list.length === 0) {
+              if (lastPos) {
+                list.push(lastPos)
+              } else {
+                list.push(start)
+              }
+            }
+            list.push(location)
+            lastPos = location
+          }
+        })
+      } else {
+        // just see if we have a position
+        if (turn.position) {
+          const ptHex = grid.hexNamed(turn.position)
+          if (ptHex) {
+            // special case - this is the first location, before we have a route
+            list.shift() // clear the start point
+            const location = ptHex.centrePos
+            list.push(location)
+            list.push(start)
+          }
+        }
+      }
+    })
+    if (boldLatLongs.length > 0) {
+      boldLine.setLatLngs(boldLatLongs)
+      res.addLayer(boldLine)
+    }
+    boldLine.setLatLngs(boldLatLongs)
+    if (feintLatLongs.length > 0) {
+      feintLine.setLatLngs(feintLatLongs)
+      res.addLayer(feintLine)
+    }
+  }
+
+  return res
+}
+
+function bearingBetween (/* latLng */ p1, /* latLng */ p2) {
+  var d2r = Math.PI / 180
+  var r2d = 180 / Math.PI
+  var lat1 = p1.lat * d2r
+  var lat2 = p2.lat * d2r
+  var dLon = (p2.lng - p1.lng) * d2r
+  var y = Math.sin(dLon) * Math.cos(lat2)
+  var x = Math.cos(lat1) * Math.sin(lat2) - Math.sin(lat1) * Math.cos(lat2) * Math.cos(dLon)
+  var brng = Math.atan2(y, x)
+  brng = parseInt(brng * r2d)
+  brng = (brng + 360) % 360
+  return brng
+};
+
+/** we plot a line marker perpendicular to the direction
+ * of travel at route waypoints.  Determine the marker to use for this bearing
+ */
+function bearingMarkerFor (/* number */ angle) {
+  let icon
+
+  if (angle === 180 || angle === 360 || angle === 0) {
```

Comment:
Is there a reason this isn't a switch?

Reply:
I thought we could only do equality tests in a switch. We need to do some < tests:
![image](https://user-images.githubusercontent.com/1108513/72913292-de932380-3d34-11ea-88f2-49c4cd299ba1.png)

Reply:
To the best of my knowledge, it's just an expression evaluator, the same as `if`

Reply:
```
switch(expression) {
  case x:
    // code block
    break;
```

To represent the above angle test in a switch, we could only do:
```
  switch(angle > 0 && angle < 90) {
    case: true:
       break
    case: false
       break
}
```
That wouldn't help :-(